src/review/review.service.ts

```diff
@@ -0,0 +1,68 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { CreateReviewDto } from './dto/create-review.dto';
+import { UpdateReviewDto } from './dto/update-review.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Review } from './entities/review.entity';
+import { Repository } from 'typeorm';
+import { Guide } from 'src/guide/entities/guide.entity';
+
+@Injectable()
+export class ReviewService {
+  constructor(
+    @InjectRepository(Review) private reviewRepository: Repository<Review>,
+    @InjectRepository(Guide) private guideRepository: Repository<Guide>,
+  ) {}
+
+  async create(createReviewDto: CreateReviewDto, userId: number) {
+    try {
+      const { guideId } = createReviewDto;
+      const guide = await this.guideRepository.findOne({
+        where: { id: guideId },
+      });
+      if (!guide) {
+        throw new BadRequestException(`Guide with ID ${guideId} not found`);
+      }
+
+      const newReview = await this.reviewRepository.create({
+        ...createReviewDto,
+        userId,
+        createdAt: new Date(),
+      });
+
+      const savedReview = await this.reviewRepository.save(newReview);
+
+      await this.guideRepository.save(guide);
+
+      return savedReview;
+    } catch (error) {
+      if (
+        error.message.includes('duplicate key value violates unique constraint')
+      ) {
+        throw new BadRequestException('user have already reviewed this guide');
+      }
+      throw new Error(error);
+    }
+  }
+
+  async findAll(guideId: number) {
+    return this.reviewRepository.find({
+      where: { guideId },
+      relations: ['user'],
+    });
+  }
```

Comment:
Again this should be retrieved from the guides list which has the reviews included with user who posted it.

If you know this is wrong then fix it, need to change the entity structure most likely.